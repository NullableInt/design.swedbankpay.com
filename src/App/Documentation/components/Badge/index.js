import React from "react";

import ComponentPreview from "../../utils/ComponentPreview";
import DocToc from "../../utils/DocToc";

const Overview = () => (
    <>
        <h2 id="overview">Overview</h2>
        <p>Overview... <code className="token property">.badge</code>...</p>
        <ComponentPreview language="html" showCasePanel={true} codeFigure={true}>
            <p>Badge <span className="badge badge-default">default</span></p>
            <p>Badge <span className="badge badge-brand">brand</span></p>
            <p>Badge <span className="badge badge-blue">blue</span></p>
            <p>Badge <span className="badge badge-purple">purple</span></p>
            <p>Badge <span className="badge badge-yellow">yellow</span></p>
            <p>Badge <span className="badge badge-red">red</span></p>
        </ComponentPreview>
    </>
);

const FurtherUsage = () => (
    <>
        <h2 id="further-usage">Further usage</h2>
        <p>Further usage... <code className="token property">.badge</code>...</p>
        <ComponentPreview language="html" showCasePanel={true} codeFigure={true}>
            <h1>Badge <small className="badge badge-default">default</small></h1>
            <h2>Badge <small className="badge badge-brand">brand</small></h2>
            <h3>Badge <small className="badge badge-blue">blue</small></h3>
            <h4>Badge <small className="badge badge-purple">purple</small></h4>
            <h5>Badge <small className="badge badge-yellow">yellow</small></h5>
            <h6>Badge <small className="badge badge-red">red</small></h6>
        </ComponentPreview>
    </>
);

const BadgeText = () => (
    <div className="col-md-12 col-lg-10 doc-body">
        <p className="lead">Badge...</p>
        <Overview />
        <FurtherUsage />
    </div>
);

const Badge = () => (
    <div className="doc-container">
        <div className="row">
            <BadgeText />
            <DocToc component={BadgeText} />
        </div>
    </div>
);

export default Badge;

/* for testing */
export { Overview, FurtherUsage, BadgeText };
