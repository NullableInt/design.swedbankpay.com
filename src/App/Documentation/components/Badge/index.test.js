import React from "react";
import { shallow } from "enzyme";

import Badge, { Overview, FurtherUsage, BadgeText } from "./index";

describe("Components: Badge", () => {
    it("is defined", () => {
        expect(Badge).toBeDefined();
    });

    it("renders", () => {
        const wrapper = shallow(
            <Badge />
        );

        expect(wrapper).toMatchSnapshot();
    });

    describe("Overview", () => {
        it("is defined", () => {
            expect(Overview).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(
                <Overview />
            );

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe("FurtherUsage", () => {
        it("is defined", () => {
            expect(FurtherUsage).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(
                <FurtherUsage />
            );

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe("BadgeText", () => {
        it("is defined", () => {
            expect(BadgeText).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(
                <BadgeText />
            );

            expect(wrapper).toMatchSnapshot();
        });
    });
});
