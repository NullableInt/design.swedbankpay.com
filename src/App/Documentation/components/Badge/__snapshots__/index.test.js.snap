// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components: Badge BadgeText renders 1`] = `
<div
  className="col-md-12 col-lg-10 doc-body"
>
  <p
    className="lead"
  >
    Badge...
  </p>
  <Overview />
  <FurtherUsage />
</div>
`;

exports[`Components: Badge FurtherUsage renders 1`] = `
<Fragment>
  <h2
    id="further-usage"
  >
    Further usage
  </h2>
  <p>
    Further usage... 
    <code
      className="token property"
    >
      .badge
    </code>
    ...
  </p>
  <ComponentPreview
    codeFigure={true}
    language="html"
    showCasePanel={true}
  >
    <h1>
      Badge 
      <small
        className="badge badge-default"
      >
        default
      </small>
    </h1>
    <h2>
      Badge 
      <small
        className="badge badge-brand"
      >
        brand
      </small>
    </h2>
    <h3>
      Badge 
      <small
        className="badge badge-blue"
      >
        blue
      </small>
    </h3>
    <h4>
      Badge 
      <small
        className="badge badge-purple"
      >
        purple
      </small>
    </h4>
    <h5>
      Badge 
      <small
        className="badge badge-yellow"
      >
        yellow
      </small>
    </h5>
    <h6>
      Badge 
      <small
        className="badge badge-red"
      >
        red
      </small>
    </h6>
  </ComponentPreview>
</Fragment>
`;

exports[`Components: Badge Overview renders 1`] = `
<Fragment>
  <h2
    id="overview"
  >
    Overview
  </h2>
  <p>
    Overview... 
    <code
      className="token property"
    >
      .badge
    </code>
    ...
  </p>
  <ComponentPreview
    codeFigure={true}
    language="html"
    showCasePanel={true}
  >
    <p>
      Badge 
      <span
        className="badge badge-default"
      >
        default
      </span>
    </p>
    <p>
      Badge 
      <span
        className="badge badge-brand"
      >
        brand
      </span>
    </p>
    <p>
      Badge 
      <span
        className="badge badge-blue"
      >
        blue
      </span>
    </p>
    <p>
      Badge 
      <span
        className="badge badge-purple"
      >
        purple
      </span>
    </p>
    <p>
      Badge 
      <span
        className="badge badge-yellow"
      >
        yellow
      </span>
    </p>
    <p>
      Badge 
      <span
        className="badge badge-red"
      >
        red
      </span>
    </p>
  </ComponentPreview>
</Fragment>
`;

exports[`Components: Badge renders 1`] = `
<div
  className="doc-container"
>
  <div
    className="row"
  >
    <BadgeText />
    <DocToc
      component={[Function]}
    />
  </div>
</div>
`;
